src/java.desktop/share/classes/com/sun/beans/introspect/MethodInfo.java

```diff
@@ -25,18 +25,33 @@
 
 package com.sun.beans.introspect;
 
+import java.io.Closeable;
+import java.io.Externalizable;
+import java.io.Serializable;
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.lang.reflect.Type;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
+import java.util.Set;
 
 import com.sun.beans.TypeResolver;
 import com.sun.beans.finder.MethodFinder;
 
 final class MethodInfo {
+
+    static final Set<Class<?>> IGNORABLE_INTERFACES = Set.of(
```

Comment:
Hm. Why only this specific interfaces?

Reply:
This a list of commonly implemented interfaces that don't need to be inspected because they are "known empty". This list is inspired by [Spring's ClassUtils](https://github.com/spring-projects/spring-framework/blob/ea83d66fb5ac7e8358015f6ad938607a96ae1b41/spring-core/src/main/java/org/springframework/util/ClassUtils.java#L820-L831). Happy to add any others that deserve to be in there.

Reply:
given that serializable is commonly implemented, this may be a worthwhile optimisation.

Reply:
I think it should have some explanation comment, about why this interfaces are ignored and how they were chosen.

Reply:
Fixed in ac90a10e238.